### ISSUE TITLE
How can I read CSV files with a single column and empty values? What about `NULL` literals?

### ISSUE THREAD

**Issue by Jehoel** [JoshClose/CsvHelper#1122]
I have a CSV file that has a single column of `Int32?` values and sometimes it only has a single row of values, so the file can look like any of these strings:

    const String csv1 = "Field1\r\n1\r\n";
    const String csv2 = "Field1\r\n\r\n";
    const String csv3 = "Field1\r\n1\r\n\r\n2\r\n";

I attempt to read like this:

    List<Int32?> values = new List<Int32?>();
    Configuration cfg = new Configuration()
    {
        IgnoreBlankLines = false
    }
    CsvReader rdr = new CsvReader( new StringReader( csv3 ), cfg );
    if( !rdr.Read() ) throw...
    if( !rdr.ReadHeader() ) throw...
    while( rdr.Read() )
    {
        Int32? value = rdr.GetField<Int32?>(0);
        values.Add( value );
    } 

Presently this code crashes with a `MissingFieldException` for `csv2` and `csv3`.

If I add a `MissingFieldFound` handler that overwrites `context.Record = new String[] { "" }` then it works as-intended. But I don't want to have to special-case the `Configuration` object for single-column CSV files. Is there a better way to handle this case?

Additionally, I have CSV files where all string values are in quotes, and instead of having an empty fields these files use a literal `NULL` to represent a NULL value - but I couldn't see any obvious way to handle this case. The files look like this:

    const String csv = "Field1,Field2,Field3\r\n1,NULL,\"a\"\r\nNULL,123,NULL";

...which should be parsed as:

    {    1, null,  "a" }
    { null,  123, null }

**Comment by Jehoel** [comment]
I've written some test-cases for this scenario and they currently fail with the current HEAD of `master` d8de6bdc.

I've found the underlying bug for my issue is that the generic `GetField<T>( Int32 )` does not have the same behaviour as `GetField( Int32 )`.

* `GetField( Int32 )` checks `context.ReaderConfiguration.IgnoreBlankLines` before calling `MissingFieldFound`.
* `GetField<T>( Int32 )` does not.

I've fixed this in my branch I'll make a PR now.
